tex/gregoriotex-spaces.tex

```diff
@@ -29,38 +29,51 @@
 %% The following macros enable users to tune penalties used in Gregorio
 
 % macro to force a break on a new line
-\def\GreForceBreak{\gre@penalty{\grenewlinepenalty}}%
+\def\GreForceBreak{\gre@penalty{\the\gre@space@count@newlinepenalty}}%
 
 % macro to prevent a line break
-\def\GreNoBreak{\gre@penalty{\grenobreakpenalty}}%
+\def\GreNoBreak{\gre@penalty{\the\gre@space@count@nobreakpenalty}}%
 
 
+
+\newcount\gre@hyphenpenaltysave%
+\newcount\gre@exhyphenpenaltysave%
+\newcount\gre@doublehyphendemeritssave%
+\newcount\gre@finalhyphendemeritssave%
+\newcount\gre@brokenpenaltysave%
+\newcount\gre@loosenesssave%
+\newcount\gre@tolerancesave%
+\newcount\gre@pretolerancesave%
+\newskip\gre@emergencystretchsave%
+\newcount\gre@widowpenaltysave%
+\newcount\gre@clubpenaltysave%
 %% The following macros cancel some useless penalties, and reinstates them
 %% at the end of a score
 
 \def\gre@cancelpenalties{%
-  \xdef\gre@hyphenpenaltysave{\the\hyphenpenalty }%
-  \xdef\gre@exhyphenpenaltysave{\the\exhyphenpenalty }%
-  \xdef\gre@doublehyphendemeritssave{\the\doublehyphendemerits }%
-  \xdef\gre@finalhyphendemeritssave{\the\finalhyphendemerits }%
-  \xdef\gre@brokenpenaltysave{\the\brokenpenalty }%
-  \hyphenpenalty=\grehyphenpenalty\relax %
-  \exhyphenpenalty=\grehyphenpenalty\relax %
+  \global\gre@hyphenpenaltysave=\hyphenpenalty\relax %
+  \global\gre@exhyphenpenaltysave=\exhyphenpenalty\relax %
+  \global\gre@doublehyphendemeritssave=\doublehyphendemerits\relax %
+  \global\gre@finalhyphendemeritssave=\finalhyphendemerits\relax %
+  \global\gre@brokenpenaltysave=\brokenpenalty\relax %
+  \global\gre@loosenesssave=\looseness\relax %
+  \global\gre@tolerancesave=\tolerance\relax %
+  \global\gre@pretolerancesave=\pretolerance\relax %
+  \global\gre@emergencystretchsave=\emergencystretch\relax %
+  \global\gre@widowpenaltysave=\widowpenalty\relax %
+  \global\gre@clubpenaltysave=\clubpenalty\relax %
+
```

Comment:
Empty, uncommented lines will be treated as spaces and may throw the spacing off.